### PR TITLE
Disable alpha to coverage when fragment shader exports sample mask

### PR DESCRIPTION
**Pull request by doitsujin**
Not immediately relevant for any game, but we've encountered this issue in D3D11, and D3D12 behaves the same.

Also proves that sample mask export still affects rendering with non-multisamples render targets.

### CODE REVIEW

File: libs/vkd3d-shader/spirv.c [hunk 1]
```diff
@@ -11813,6 +11827,56 @@ void vkd3d_shader_extract_feature_meta(struct vkd3d_shader_code *code)
                 meta |= VKD3D_SHADER_META_FLAG_EMITS_TRIANGLES;
 
         }
+        else if ((op == SpvOpDecorate && count == 4) ||
+                (op == SpvOpMemberDecorate && count == 5))
```
Comment (HansKristian-Work) [inline libs/vkd3d-shader/spirv.c:11831]: Member decorate? SpvBuiltInSampleMask cannot be placed in IO block.

File: libs/vkd3d-shader/spirv.c [hunk 2]
```diff
@@ -11813,6 +11827,56 @@ void vkd3d_shader_extract_feature_meta(struct vkd3d_shader_code *code)
...
+
+            if (storage_class == SpvStorageClassOutput || storage_class == SpvStorageClassInput)
+            {
+                var_id = spirv[offset + 2];
+
+                for (i = 0; i < tracked_builtin_count; i++)
+                {
+                    const struct vkd3d_tracked_builtin *entry = &tracked_builtins[i];
+
+                    if (entry->var_id != var_id)
```
Comment (HansKristian-Work) [inline libs/vkd3d-shader/spirv.c:11865]: Is it written like this to potentially look for more than builtin later?

File: libs/vkd3d-shader/spirv.c (hunk 1 above)
Comment (HansKristian-Work) [reply libs/vkd3d-shader/spirv.c:11831]: Is the intention to be more extensible here?

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
Nits/questions

File: libs/vkd3d-shader/spirv.c (hunk 1 above)
Comment (doitsujin) [reply libs/vkd3d-shader/spirv.c:11831]: Pretty much, yes.

File: libs/vkd3d-shader/spirv.c (hunk 2 above)
Comment (doitsujin) [reply libs/vkd3d-shader/spirv.c:11865]: The idea here was to introduce a mechanism which can be easily extended in the future. I think there's value to that, e.g. if we ever upgrade one of the shader compilers to SPIR-V 1.6 and things like viewport/layer export change, and potential future changes.

**Review by HansKristian-Work — APPROVED** [review]
LGTM

**Pull request merged**